content/en/docs/getting-started/reference-application-specification.md

```diff
@@ -0,0 +1,119 @@
+  builds and runs in both versions.
+- The application must be runnable from a command line interface.
+- There must be a Dockerfile to run the application within a containerized
+  environment.
+- The application must run standing alone. In other words, it cannot take hard
+  dependencies on any other content of the repository enclosing it. This allows
+  users to copy the code into their own project without untangling the
+  dependency management.
+
+## Service requirements
+
+- The application must listen at port 8080 for HTTP requests by default. The
+  port should be configurable via the environment variable APPLICATION_PORT.
+- For handling the HTTP requests, a library should be used for which an
+  instrumentation library is available. The application must provide the
+  endpoint `/rolldice?rolls=<n>` via `GET` (and optionally `POST`) and return
+  the following HTTP status codes and JSON results:
+  - if `rolls` is not set or has a valid input (positive integer > 0): status
+    code `200` and either a single number between 1 and 6 if `rolls` is not set
+    or 1, or an array of `n` numbers between 1 and 6 where `n` is the value of
+    `rolls`
+  - if `rolls` is set to an invalid input (not a number): status code `400` and
+    `{"status": "error", "message": "Parameter rolls must be a positive integer"}`
+  - if `rolls` is set to a `0` or a negative integer: status code `500` and no
+    JSON output. The error examples will be used to demonstrate how
+    OpenTelemetry can be used to identify errors.
+- There may be an optional attribute `player=name` to the `/rolldice` endpoint.
+- The application must output the following log lines:
+  - an INFO-level message for each HTTP request with a status code `<400`
+  - a WARN-level message for each HTTP request with a status code between `400`
+    and `499`, including the message which will be sent in the JSON result
+  - a ERROR-level message for each HTTP request with a status code above `499`
+  - if the optional `player` attribute is set, a DEBUG-level message that
+    outputs the value of `player` and the rolled number
+  - if the optional `player` attribute is not set, a DEBUG-level message that
+    outputs the static value `anonymous player` and the rolled number. The log
+    lines will be used to add a log bridge during instrumentation to demonstrate
+    how OpenTelemetry can connect to existing logging frameworks.
+- The code of the application must be split into two files:
+  - an `app` file that contains the handling of the HTTP requests
+  - a `library` file that contains the implementation of the roll dice function.
+    The names of those files should be idiomatic within the implementing
+    language, like `app.js` and `roll-the-dice.js`. The important point is that
+    by separating the two, the code demonstrates that the `library` only takes
+    dependency on the API and all the SDK code is initalized in the `app` code.
```

Comment:
@martincostello @cijothomas is this better, I picked a `.js` example but if you provide me with what a good naming for .NET is, I can update that accordingly.